moto/rds/models.py

```diff
@@ -586,6 +588,22 @@ def __init__(
         )
         self.parameter_group = db_cluster_parameter_group_name or default_pg
         self.db_subnet_group_name = db_subnet_group_name or "default"
+        self.domain_membership: Optional[dict[str, str | list[str] | None]] = (
+            {
+                "Domain": domain.split(".")[0] if domain else None,
+                "Status": "active",
+                "FQDN": domain if domain else None,
+                "IAMRoleName": domain_iam_role_name
+                or "rds-directory-service-access-role",
+                "OU": f"OU={domain.split('.')[0]}OU,DC={domain.split('.')[0]},DC=com"
+                if domain
+                else None,
+                "AuthSecretArn": None,  # Not implemented yet
+                "DnsIps": [],
+            }
+            if domain
+            else None
+        )
```

Comment:
How about encapsulating this as a DomainMembership object that takes in the relevant params (`domain`, etc.)?  I'd really like to get away from dicts when we know the shape of the object/attributes ahead of time.